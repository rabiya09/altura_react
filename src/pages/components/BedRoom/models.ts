export interface BedRoomProps {
    title: string;
    description: string;
    readmore: string;
}