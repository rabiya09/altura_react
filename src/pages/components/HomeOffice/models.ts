export interface HomeOfficeProps {
    title: string;
    description: string;
    readmore: string;
}