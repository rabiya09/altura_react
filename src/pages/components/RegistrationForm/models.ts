export interface FieldStructure {
    dirty: boolean; 
    error: boolean; 
    message: string;
}
  
  export interface FieldStr {
    email: FieldStructure;
    password: FieldStructure;
    confirmPassword: FieldStructure;
  }

  export interface SignUpFormState  {
    firstname: string;
    lastname: string;
    email: string;
    password: string
  }
  type Gender = 'male' | 'female' | 'non-binary';
  export interface User {
    name: string;
    age: number;
    email: string;
    gender: Gender;
    password: string;
  }