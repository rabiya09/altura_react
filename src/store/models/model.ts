export interface AddToCartState {
    items: [
        {
            id: 0;
            name: string; 
            description: string; 
            price:  string; 
            imageUrl: string; 
            quantity: number;
        },       
    ];
    addedItems:[
        {
            id: 0;
            name: string; 
            description: string; 
            price:  string; 
            imageUrl: string; 
            qtyInCart: number;
            quantity: number;
        }, 
    ];
    total: 0;   
    invoiceDate: Date;
    invoiceNumber: string; 
  }