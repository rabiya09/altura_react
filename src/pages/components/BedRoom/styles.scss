.bed-room-container {
    width: 100%;
    margin: 0 auto; 
    padding-bottom: 100px;
    float: left;
    @media (max-width: 599px) {
        padding-bottom: 10px;
    }
    .bed-room {
        width: 94%;
        margin: 0 auto;
        margin-left: 4%;
        float: left;
        .bed-room-text {
            float: left;
            width: 30%;
            margin-left: 5%;
            margin-top: 175px;
            .read-more {
                width: 100%;
                float: left;
                padding-top: 10px;
                a {
                    color: #E41C77;
                    text-decoration: none;
                }
            }
            @media (max-width: 599px) {
                width: 100%;
                text-align: center;  
                margin-top: 10px;                  
            }
        }
        .bed-room-title {
            font-weight: bold;
            float: left;
            font-size: 18px;
            width: 100%;
           padding-bottom: 10px;
        }
        .bed-room-image {
            float: left;
            width: 58%;
            @media (max-width: 599px) {
                width: 100%;
            }
            img {
                @media (max-width: 599px) {
                    width: 356px;                     
                }
            }
        }
    }
}