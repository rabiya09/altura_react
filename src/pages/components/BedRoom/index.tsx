import './styles.scss';
import { Link } from "react-router-dom";
import bedRoom from '../../../assets/images/bedroom.svg';
import { BedRoomProps } from './models';

const BedRoom = (props: BedRoomProps) => {
    return (
        <div className="bed-room-container">
    <div className="bed-room">  
        <div className="bed-room-image">
            <img alt="Bed Room" src={bedRoom}/>
        </div>      
        <div className="bed-room-text">
            <span className="bed-room-title" data-testid="page-title">{props.title}</span>
            {props.description}
            <span className="read-more">
                <Link to='/shop-all-bed-room'>{props.readmore}</Link>
            </span>
        </div>        
    </div>
    </div>
    );

}

export default BedRoom;