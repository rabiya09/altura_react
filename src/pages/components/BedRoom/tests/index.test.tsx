import { render, screen } from '@testing-library/react';
import BedRoom from '../index';
import { MemoryRouter } from 'react-router-dom';
import { getProps } from './__fakes__';
import { BedRoomProps } from '../models';

describe('BedRoom', () => {
    const props = getProps();
    test('should match snapshot', () => {
      const {asFragment} = render(<MemoryRouter><BedRoom {...getProps() as BedRoomProps}/></MemoryRouter>);
      expect(screen.getByTestId('page-title')).toBeInTheDocument();      
      expect(asFragment()).toMatchSnapshot();
    })  
  });