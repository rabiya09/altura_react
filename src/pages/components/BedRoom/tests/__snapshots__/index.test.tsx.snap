// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedRoom should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="bed-room-container"
  >
    <div
      class="bed-room"
    >
      <div
        class="bed-room-image"
      >
        <img
          alt="Bed Room"
          src="bedroom.svg"
        />
      </div>
      <div
        class="bed-room-text"
      >
        <span
          class="bed-room-title"
          data-testid="page-title"
        >
          Bed Room
        </span>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        <span
          class="read-more"
        >
          <a
            href="/shop-all-bed-room"
          >
            Shop All
          </a>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
